### PR TITLE
add rabin min error test

### PR DESCRIPTION
**Pull request by kjzz**
feature #5425 

License: MIT
Signed-off-by: Kejie Zhang <601172892@qq.com>

### CODE REVIEW

**Comment by kjzz** [comment]
@Stebalien after you update gx dep , i add tests about rabin min error.please help me review it ,Thx

File: test/sharness/t0040-add-and-cat.sh [hunk 1]
```diff
@@ -156,6 +156,15 @@ test_add_cat_file() {
     test_cmp expected actual
   '
 
+  test_expect_success "ipfs add --chunker rabin-36-512-1024 succeeds" '
+    ipfs add --chunker rabin-36-512-1024 mountdir/hello.txt >actual
+  '
+
+  test_expect_failure "ipfs add --chunker rabin-12-512-1024 failed" '
```
Comment (Stebalien) [inline test/sharness/t0040-add-and-cat.sh]: FYI, `test_expect_failure` is actually poorly named. It's for testing known bugs, not testing things that *should* fail. This should actually just be `test_expect_code 1 add --chunker ...`

File: test/sharness/t0040-add-and-cat.sh [hunk 2]
```diff
@@ -156,6 +156,15 @@ test_add_cat_file() {
     test_cmp expected actual
   '
 
+  test_expect_success "ipfs add --chunker rabin-36-512-1024 succeeds" '
+    ipfs add --chunker rabin-36-512-1024 mountdir/hello.txt >actual
```
Comment (Stebalien) [inline test/sharness/t0040-add-and-cat.sh]: Given that we're not actually *checking* these results, we should probably just run this command as `ipfs add -Q` (quiet) and *not* redirect to `actual`. We redirect to a file when comparing output with some expected output.

**Comment by kjzz** [comment]
hey @Stebalien ,Thx a lot for your advice. I have update my test.but maybe something wrong with ci.Can u help me review it?Thx.

**Comment by Stebalien** [comment]
Windows CI broke yesterday. I'm rerunning it now.

File: test/sharness/t0040-add-and-cat.sh [hunk 3]
```diff
@@ -156,6 +156,15 @@ test_add_cat_file() {
     test_cmp expected actual
   '
 
+  test_expect_success "ipfs add --chunker rabin-36-512-1024 succeeds" '
+    ipfs add -Q --chunker rabin-36-512-1024 mountdir/hello.txt
+  '
+
+  test_expect_code 127 "ipfs add --chunker rabin-12-512-1024 failed" '
```
Comment (Stebalien) [inline test/sharness/t0040-add-and-cat.sh]: Ah, sorry, my comment was confusing. This needs to be:

```
test_expect_success "...." '
    test_expect_code 127 ...
'
```

(or something like that).

Comment (Stebalien) [reply test/sharness/t0040-add-and-cat.sh]: (fixed)

**Comment by kjzz** [comment]
@Stebalien Thx for help me fix it.

**Comment by Stebalien** [comment]
Np. Small nits like that are often easier to just fix quickly instead of wasting your time going back and forth.

**Comment by kjzz** [comment]
Hey @Stebalien , if If you have time，can u give me some advice about issue #4442 and #5432 .And I want to contribute more in ipfs.or if you any issue that i can help you,please ping me.Thx a lot.

**Pull request merged**